internal/prometheusbp/spectest/spec.go

```diff
@@ -0,0 +1,190 @@
+package spectest
+
+import (
+	"errors"
+	"fmt"
+	"strings"
+	"testing"
+
+	"github.com/google/go-cmp/cmp"
+	"github.com/prometheus/client_golang/prometheus"
+	"github.com/prometheus/client_golang/prometheus/testutil"
+
+	"github.com/reddit/baseplate.go/errorsbp"
+)
+
+const (
+	client = "client"
+	server = "server"
+)
+
+var (
+	errPrefix         = errors.New("the prefix is not at the beginning of the metric name")
+	errClientServer   = fmt.Errorf("missing %q or %q as the second part of the metric name", client, server)
+	errLength         = errors.New("metric name should have a minimum of three parts")
+	errNotFound       = errors.New("metric not found")
+	errPrometheusLint = errors.New("prometheus GatherAndLint")
+	errDiffLabels     = errors.New("labels are incorrect")
+)
+
+// ValidateSpec validates that the Prometheus metrics being exposed from baseplate.go
+// conform to the baseplate spec.
+func ValidateSpec(t *testing.T, metricPrefix string, wantMetricCount int) {
```

Comment:
🔕 ah sorry I mentioned this deeper in the call stack and didn't explain that I was imagingin that it would make it up this far.

I kinda imagine that someone might want to use this independently to check client and server metrics -- e.g. when only one of them is implemented 😅 

so I would probably make this API

```go
func ValidateSpec(t *testing.T, metricPrefix, clientOrServer string)
```

As mentioned earlier, the count might be confusing because the spec already catches missing metrics, and this way you can say `ValidateSpec("grpc", "server")` instead of `"grpc_server"` which has a bit more "magic" to it.

Reply:
will do. also forgot to remove the `wantMetricCount` arg. will fix.